### DISCUSSION TITLE
Example of annotation ?

### DISCUSSION THREAD

**Discussion by abhishek-1985** [grafana/grafonnet#93] [Q&A] [answered]
Hi,

I would like to add annotations to my dashboard and was wondering if you help me with an example for creating the annotation object and how it should be passed into dashboard ?

Also can I create a new annotation object using `g.annotation.new` ?

**Comment by Duologic** [comment, marked as answer]
The annotations part of grafonnet has not received a lot of attention yet. I just had a quick look at how to do this with the current code but it not very user-friendly yet.

```jsonnet
local grafonnet = import 'github.com/grafana/grafonnet/gen/grafonnet-latest/main.libsonnet';
local annotation = g.dashboard.annotation;

your_dashboard
+ g.dashboard.withAnnotations([
  annotation.datasource.withType('prometheus')
  + annotation.datasource.withUid('default')
  + annotation.withName('A vector annotation')
  + {
    // adding these here as the upstream schema does not implement them yet, so there is not function for it
    expr: 'vector(1)',
    enabled: true,
  },
])
```
